app/models/hardware.rb

```diff
@@ -44,6 +43,11 @@ def mac_addresses
   def ram_size_in_bytes
     memory_mb.to_i * 1.megabyte
   end
+  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
+  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Multiplication.new(
+      Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]), 1.megabyte))
+  end)
```

Comment:
As an alternative solution I have migration PR (not merged) to change datatype of ```memory_mb``` to ```bigint``` and on my test db that column was ```bigint```. So, I was struggling with providing test which is failing on master  ....